extensions/wikia/WikiFactory/api/MarkWikiAsClosedController.class.php

```diff
@@ -21,18 +23,40 @@ public function markWikiAsClosed() {
 		$wikiId = $request->getVal( self::WIKI_ID );
 		$reason = $request->getVal( self::REASON );
 		$userId = $request->getVal( self::USER_ID );
+		$fandomCreatorCommunityId = $request->getVal( self::FANDOM_CREATOR_COMMUNITY_ID );
 
 		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
 
-		if ( !is_numeric( $wikiId ) || empty( $reason ) || !is_numeric( $userId)) {
+		if ( !empty( $fandomCreatorCommunityId ) && !is_numeric( $fandomCreatorCommunityId ) ) {
+			$this->response->setCode( 400 );
+			$this->info('invalid  fandomCreatorCommunityId parameter in request');
+		} else if ( !is_numeric( $wikiId ) || empty( $reason ) || !is_numeric( $userId ) ) {
 			// No wikiId, userId or reason given: Bad Request
 			$this->response->setCode( 400 );
 			$this->info('no wikiId or reason parameter in request');
 		} else {
 			$user = User::newFromId($userId);
-			$res = WikiFactory::setPublicStatus( WikiFactory::CLOSE_ACTION, $wikiId, $reason, $user);
+			$goForClose = true;
```

Comment:
How do you feel about extracting this to a private method? Something like `shouldGoForClose`

Reply:
Agreed, done.